### PR TITLE
Basic Word Relevancy Testing

### PR DESCRIPTION
**Pull request by Tytynguyen**
Added basic word relevancy testing using definition similarities, part of speech similarities, shared synonyms, and antonyms. This roughly converts the similarities into a %, using a sigmoid function. The weights were manually set, and roughly tested

### CODE REVIEW

File: TextAnalysis/src/langanal/word/processing/WordProcessing.java [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+						System.out.println("\tantonym: " + antonymValue);
+					}
+					
+					relevancy += (definitionValue + POSValue + synonymValue + antonymValue);
+
+				}
+			}
+			relevancy = (float) (200*(1/(1+Math.pow(Math.E,-(relevancy/5)))-0.5));
+		}else{
+			System.out.println("ERROR: One of the words was not found in the dictionary or thesaurus. Check your spelling.");
```
Comment (jacobstein) [inline TextAnalysis/src/langanal/word/processing/WordProcessing.java]: Should not print out to consol

File: TextAnalysis/src/langanal/word/processing/WordProcessing.java [hunk 2]
```diff
@@ -39,18 +41,26 @@ public static float compareWords(String word1, String word2){
...
+						System.out.println("\tantonym: " + antonymValue);
+					}
+					
 					relevancy += (definitionValue + POSValue + synonymValue + antonymValue);
+
 				}
 			}
+			relevancy = (float) (200*(1/(1+Math.pow(Math.E,-(relevancy/5)))-0.5));
 		}else{
 			System.out.println("ERROR: One of the words was not found in the dictionary or thesaurus. Check your spelling.");
```
Comment (edelmanjm) [inline TextAnalysis/src/langanal/word/processing/WordProcessing.java]: Change to System.err.println

File: TextAnalysis/src/langanal/word/processing/WordProcessing.java [hunk 3]
```diff
@@ -39,18 +41,26 @@ public static float compareWords(String word1, String word2){
...
+						System.out.println("\tPOS: " + POSValue);
+						System.out.println("\tsynonym: " + synonymValue);
+						System.out.println("\tantonym: " + antonymValue);
+					}
+					
 					relevancy += (definitionValue + POSValue + synonymValue + antonymValue);
+
 				}
 			}
+			relevancy = (float) (200*(1/(1+Math.pow(Math.E,-(relevancy/5)))-0.5));
```
Comment (edelmanjm) [inline TextAnalysis/src/langanal/word/processing/WordProcessing.java:66]: What does this math do? Add a comment explaining why the math is the way it is.

File: TextAnalysis/src/langanal/word/processing/WordProcessing.java [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+	 */
+	public static float compareWords(String word1, String word2){
+		float relevancy = 0;	//in %
+
+		LinkedList<Word> allWord1 = new LinkedList<Word>();
+		LinkedList<Word> allWord2 = new LinkedList<Word>();
+		try {
+			allWord1 = WordInfo.getFullInfoWords(word1);
+			allWord2 = WordInfo.getFullInfoWords(word2);
+		} catch (Exception e) {
```
Comment (edelmanjm) [inline TextAnalysis/src/langanal/word/processing/WordProcessing.java]: Why would an exception be thrown? Why are we handling it this way? Comment.

File: TextAnalysis/src/langanal/word/processing/WordProcessing.java [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+	 * Compares, processes, and returns the relevancy of the two words
+	 * 
+	 * @param word1
+	 * @param word2
+	 * @return The % relevancy
+	 */
+	public static float compareWords(String word1, String word2){
+		float relevancy = 0;	//in %
+
+		LinkedList<Word> allWord1 = new LinkedList<Word>();
```
Comment (edelmanjm) [inline TextAnalysis/src/langanal/word/processing/WordProcessing.java:35]: Unclear initially on what the allWord s are for.

Comment (edelmanjm) [inline TextAnalysis/src/langanal/word/processing/WordProcessing.java:35]: Would an array list be better? There don't seem to be a ton of writes, but there are a lot of reads.

File: TextAnalysis/src/langanal/word/processing/WordProcessing.java [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+				if(curSynonymWord1.equals(curSynonymWord2)){
+					repeatedSynonyms++;
+				}
+			}
+		}
+		return repeatedSynonyms;
+	}
+
+	/**
+	 * Cleans and formats the ArrayList for the checkSynonyms function
```
Comment (edelmanjm) [inline TextAnalysis/src/langanal/word/processing/WordProcessing.java]: Little unclear how's it's actually cleaning, a little more depth would be nice.

**Comment by jacobstein** [comment]
LGTM

File: TextAnalysis/src/langanal/word/processing/WordProcessing.java [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+	 * @param list
+	 * @return cleaned and formatted list
+	 */
+	private static ArrayList<String> cleanSynonyms(LinkedList<String> list){
+		ArrayList<String> returnList = new ArrayList<String>();
+
+		for(String s : list){
+			for(String x : s.split(" ")){
```
Comment (edelmanjm) [inline TextAnalysis/src/langanal/word/processing/WordProcessing.java]: Minor, but how does this perform on compound words that lack a hyphen? Does it split the words?

**Pull request merged**